From b4581d8f5ba5e48de9a97988d95739b364222acd Mon Sep 17 00:00:00 2001
From: Josh Stone <jistone@redhat.com>
Date: Thu, 30 Jan 2020 22:02:59 -0800
Subject: [PATCH] Use `cargo install --no-track` with cargo 1.41

In cargo 1.41, `cargo install` adds a new `.crates2.json` file in the
install root for tracking version upgrades. This would be another file
that `%cargo_install` should remove afterward, but there is also a new
`--no-track` option which disables such metadata files altogether.

This update should be coordinated with the Rust toolchain update in the
distro, e.g. with rust-packaging `Requires: cargo >= 1.41`. The new
option will be rejected as an unstable option on earlier versions.
---
 data/macros.cargo | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/macros.cargo b/data/macros.cargo
index 5f60111..a916699 100644
--- a/data/macros.cargo
+++ b/data/macros.cargo
@@ -101,11 +101,11 @@ fi \
     %{shrink:\
       %{__cargo} install                                            \
         %{__cargo_common_opts}                                      \
+        --no-track                                                  \
         --path .                                                    \
         %{__cargo_parse_opts %{-n} %{-a} %{-f:-f%{-f*}}}            \
         %*                                                          \
     } \
-    %{__rm} %{buildroot}%{_prefix}/.crates.toml                     \
   fi \
 %endif \
 )
-- 
2.26.0

